AODN/MHL/process_MHLwave_from_txt.py

```diff
@@ -0,0 +1,305 @@
+    WHTE[:] = data['H10'].values
+    WMXH[:] = data['Hmax'].values
+    TCREST[:] = data['Tc'].values
+    VAVT[:] = data['Tz'].values
+    WPTH[:] = data['Tsig'].values
+    YRMS[:] = data['Yrms'].values
+    WPPE[:] = data['TP1'].values
+    TP2[:] = data['TP2'].values
+    M0[:] = data['M0'].values
+    VDIR[:] = data['WDIR'].values
+
+    ncfile.close()
+
+
+def convert_to_utc(timestamp, format):
+    """
+    - Convert local timestamp to UTC without taking
+    Daylight Saving Time into account. Fixed time difference of 10 hours
+    - Then generate an array of timestamp converted to
+     decimal time, IMOS format: days since 1950-01-01T00:00:00Z
+     input: timestamp string '%d/%m/%Y %H:%M:%S'
+           format : string  'old' or 'new'
+     output:
+     - time:array of decimal time from 1950-01-01T00:00:00Z
+     - dtime: array of datetime object
+    """
+    epoch = datetime(1950, 1, 1)
+    time = []
+    dtime = []
+    for t in timestamp:
+        if format == 'old':
+            dt = datetime.strptime(t, '%d/%m/%Y %H:%M:%S')
+        elif format == 'new':
+            assert type(t) is str, pdb.set_trace()
+            dt = datetime.strptime(t, '%d-%b-%Y %H:%M')
+
+        dt = dt - timedelta(hours=10)
+        dtime.append(dt)
+        time.append((dt - epoch).total_seconds())
+
+    time = np.array(time) / 3600. / 24.
+    dtime = np.array(dtime)
+    return (dtime, time)
+
+
+def create_netcdf_filename(site_code, data_array, dtime):
+    """
+    Create IMOS compliant filename
+    """
+    start_date = dtime[0].strftime('%Y%m%dT%H%M%SZ')
+    end_date = dtime[-1].strftime('%Y%m%dT%H%M%SZ')
+    return "IMOS_ANMN-NSW_W_%s_%s_FV01_END-%s" % (
+        start_date, site_code, end_date)
+
+
+def parse_nc_attribute(input_netcdf_file_path, output_nc_obj):
+    """
+    Read in attributes from a netcdf filename
+    and return attribute
+    gatts, data, annex = parse_nc_attribute(netcdf_file_path)
+    """
+    print "reading attributes from %s" % input_netcdf_file_path
+    input_nc_obj = Dataset(
+        input_netcdf_file_path, 'r', format='NETCDF3_CLASSIC')
+    output_nc_obj.title = input_nc_obj.title
+    output_nc_obj.institution = input_nc_obj.dataContact_organisationName
+    output_nc_obj.abstract = input_nc_obj.abstract
+    output_nc_obj.keywords = input_nc_obj.keywords
+    output_nc_obj.principal_investigator = (input_nc_obj.
+                                            principal_investigator)
+    output_nc_obj.cdm_data_type = input_nc_obj.cdm_data_type
+    output_nc_obj.platform_code = input_nc_obj.platform_code
+    output_nc_obj.comment = input_nc_obj.metadataDataQualityLineage
+
+    input_nc_obj.close()
+
+
+if __name__ == '__main__':
+
+    # read filename from command line
+    # if len(sys.argv) < 2:
+    #        print >>sys.stderr, 'No filename specified!'
+    #        exit(1)
+
+    # function = sys.argv[1]
+    # txtfile = sys.argv[2]
+    for txtfile in glob.glob('/vagrant/tmp/MHL/TXTFILES/*.TXT'):
```

Comment:
this script is not called by the pipeline but run locally. It's been running fine so far and i think it will run fine in the future. Hopefully.